src/extension_commands.ml

```diff
@@ -158,6 +158,71 @@ let _open_ocamllsp_output_pane, _open_ocaml_platform_ext_pane, _open_ocaml_comma
       (handler Output.command_output_channel) )
 ;;
 
+module Decorations = struct
+  let highlighting_decoration =
+    let options =
+      DecorationRenderOptions.create
+        ~backgroundColor:
+          (`ThemeColor (ThemeColor.make ~id:"editorHoverWidget.foreground"))
+        ~color:(`ThemeColor (ThemeColor.make ~id:"editorHoverWidget.background"))
+        ~border:"1px solid"
+        ~borderColor:(`ThemeColor (ThemeColor.make ~id:"editorHoverWidget.border"))
+        ~isWholeLine:false
+        ()
+    in
+    Window.createTextEditorDecorationType ~options
+  ;;
+
+  let highlight_range text_editor range =
+    let _decorationOptions =
+      let renderOptions =
+        let before = ThemableDecorationAttachmentRenderOptions.create () in
+        let options = ThemableDecorationInstanceRenderOptions.create ~before () in
+        Some (DecorationInstanceRenderOptions.create ~light:options ~dark:options ())
+      in
+      DecorationOptions.create ~range ~renderOptions ()
+    in
+    TextEditor.setDecorations
+      text_editor
+      ~decorationType:highlighting_decoration
+      ~rangesOrOptions:(`Ranges [ range ])
+  ;;
+
+  let remove_all_highlights text_editor =
+    let rangesOrOptions = `Ranges [] in
+    TextEditor.setDecorations
+      text_editor
+      ~decorationType:highlighting_decoration
+      ~rangesOrOptions
+  ;;
+end
+
+let show_selection selection text_editor =
+  TextEditor.revealRange
+    text_editor
+    ~range:(Selection.to_range selection)
+    ~revealType:TextEditorRevealType.InCenterIfOutsideViewport
+    ()
+;;
+
+(**  If the user hits the ESC key, this should go back to the initial_selection,
+     otherwise the current position of the click is used *)
+let move_cursor_after_selection_change event_fired () =
+  let onDidChangeTextEditorSelection_listener event =
+    event_fired := true;
+    let selections = TextEditorSelectionChangeEvent.selections event in
+    match selections with
+    | [ selection ] ->
+      show_selection selection (TextEditorSelectionChangeEvent.textEditor event)
+    | _ -> ()
```

Comment:
In my view of the control flow, we don't need to do anything here, simply update the flag that says the selection changed.